juriscraper/pacer/email.py

```diff
@@ -45,3 +52,71 @@ def _get_date_filed(self):
         return convert_date_string(
             date_filed[0].lower().replace("filed on ", "")
         )
+
+    def _get_document_number(self):
+        document_number = self._get_document_number_from_link()
+        return (
+            document_number
+            if document_number is not None
+            else self._get_document_number_from_text()
+        )
+
+    def _get_document_number_from_link(self):
+        try:
+            path = '//td[contains(., "Document Number:")]/following-sibling::td[1]/a/text()'
+            return numbers_only(self.tree.xpath(path)[0])[0]
+        except IndexError:
+            return None
+
+    def _get_document_number_from_text(self):
+        try:
+            path = '//td[contains(., "Document Number:")]/following-sibling::td[1]/text()'
+            return numbers_only(self.tree.xpath(path)[0])[0]
+        except IndexError:
+            return None
+
+    def _get_document_url(self):
+        try:
+            path = '//td[contains(., "Document Number:")]/following-sibling::td[1]/a/@href'
+            return self.tree.xpath(path)[0]
+        except IndexError:
+            return None
+
+    def _get_docket_entries(self):
+        try:
+            path = '//strong[contains(., "Docket Text:")]/following-sibling::font[1]/b/text()'
+            description = clean_string(self.tree.xpath(path)[0])
+            return [
+                {
+                    "date_filed": self._get_date_filed(),
+                    "description": description,
+                    "document_number": self._get_document_number(),
+                    "document_url": self._get_document_url(),
+                    "pacer_doc_id": None,
+                    "pacer_seq_no": None,
+                }
+            ]
+        except IndexError:
+            return []
+
+    def _get_email_recipients(self):
+        path = '//b[contains(., "Notice has been electronically mailed to")]/following-sibling::text()'
+        recipient_lines = self.tree.xpath(path)
+        email_recipients = []
+        for line in recipient_lines:
+            if "@" in line:
+                clean_line = clean_string(line)
```

Comment:
Do you need this line if you clean the subparts in the next couple lines?

Reply:
No sir. Removed it.